pkg/console/subresource/configmap/configmap.go

```diff
@@ -40,54 +36,62 @@ func getApiUrl(infrastructureConfig *configv1.Infrastructure) string {
 	return ""
 }
 
-// DefaultConfigMap returns
-// - a new configmap,
-// - a bool indicating if config was merged (unsupportedConfigOverrides)
-// - an error
-func DefaultConfigMap(operatorConfig *operatorv1.Console, consoleConfig *configv1.Console, managedConfig *corev1.ConfigMap, infrastructureConfig *configv1.Infrastructure, rt *routev1.Route) (*corev1.ConfigMap, bool, error) {
-
-	// Build a default config that can be overwritten from other sources
-	host := rt.Spec.Host
-	apiServerURL := getApiUrl(infrastructureConfig)
-	defaultProviders := operatorv1.ConsoleProviders{}
-	defaultConfig := NewYamlConfig(host, defaultLogoutURL, DEFAULT_BRAND, DEFAULT_DOC_URL, apiServerURL, defaultProviders)
-	// Get console config from the openshift-config-managed namespace
+func statusPageId(operatorConfig *operatorv1.Console) string {
+	if operatorConfig.Spec.Providers.Statuspage != nil {
+		return operatorConfig.Spec.Providers.Statuspage.PageID
+	}
+	return ""
+}
+
+func DefaultConfigMap(
+	operatorConfig *operatorv1.Console,
+	consoleConfig *configv1.Console,
+	managedConfig *corev1.ConfigMap,
+	infrastructureConfig *configv1.Infrastructure,
+	rt *routev1.Route) (consoleConfigmap *corev1.ConfigMap, unsupportedConfigMerged bool, err error) {
+
+	b := &ConsoleYAMLConfigBuilder{}
+	defaultConfig, err := b.Host(rt.Spec.Host).
+		LogoutURL(defaultLogoutURL).
+		Brand(DEFAULT_BRAND).
+		DocURL(DEFAULT_DOC_URL).
+		APIServerURL(getApiUrl(infrastructureConfig)).
+		ConfigBytes()
+
 	extractedManagedConfig := extractYAML(managedConfig)
 
-	// Derive a user defined config using operator config
-	logoutRedirect := consoleConfig.Spec.Authentication.LogoutRedirect
-	brand := operatorConfig.Spec.Customization.Brand
-	docURL := operatorConfig.Spec.Customization.DocumentationBaseURL
-	providers := operatorConfig.Spec.Providers
-	userDefinedConfig := NewYamlConfig(host, logoutRedirect, brand, docURL, apiServerURL, providers)
+	b2 := &ConsoleYAMLConfigBuilder{}
+	userDefinedConfig, err := b2.Host(rt.Spec.Host).
+		LogoutURL(consoleConfig.Spec.Authentication.LogoutRedirect).
+		Brand(operatorConfig.Spec.Customization.Brand).
+		DocURL(operatorConfig.Spec.Customization.DocumentationBaseURL).
+		APIServerURL(getApiUrl(infrastructureConfig)).
+		StatusPageID(statusPageId(operatorConfig)).
+		ConfigBytes()
 
 	unsupportedConfigOverride := operatorConfig.Spec.UnsupportedConfigOverrides.Raw
-
-	// merge configs with overrides, if we have them
-	mergedConfig, err := resourcemerge.MergeProcessConfig(nil, defaultConfig, extractedManagedConfig, userDefinedConfig, unsupportedConfigOverride)
-	if err != nil {
-		logrus.Errorf("failed to merge configmap: %v \n", err)
-		return nil, false, err
+	willMergeConfigOverrides := len(operatorConfig.Spec.UnsupportedConfigOverrides.Raw) != 0
```

Comment:
why not use the new var ?
```suggestion
	willMergeConfigOverrides := len(unsupportedConfigOverride) != 0
```

Reply:
will do.